extensions/notebook/src/dialog/pythonPathLookup.ts

```diff
@@ -0,0 +1,161 @@
+				'/opt/*conda*/bin/python3',
+				'/usr/share/*conda*/bin/python3',
+				`${userFolder}/*conda*/bin/python3`
+			];
+		} else {
+			let userFolder = process.env['USERPROFILE'].replace('\\', '/').replace('C:', '');
+			this.condaLocations = [
+				'/ProgramData/[Mm]iniconda*/python.exe',
+				'/ProgramData/[Aa]naconda*/python.exe',
+				`${userFolder}/[Mm]iniconda*/python.exe`,
+				`${userFolder}/[Aa]naconda*/python.exe`,
+				`${userFolder}/AppData/Local/Continuum/[Mm]iniconda*/python.exe`,
+				`${userFolder}/AppData/Local/Continuum/[Aa]naconda*/python.exe`
+			];
+		}
+	}
+
+	public async getSuggestions(): Promise<PythonPathInfo[]> {
+		let pythonSuggestions = await this.getPythonSuggestions();
+		let condaSuggestions = await this.getCondaSuggestions();
+
+		if (pythonSuggestions) {
+			if (condaSuggestions && condaSuggestions.length > 0) {
+				pythonSuggestions = pythonSuggestions.concat(condaSuggestions);
+			}
+			return this.getInfoForPaths(pythonSuggestions);
+		} else {
+			return [];
+		}
+	}
+
+	private async getCondaSuggestions(): Promise<string[]> {
+		try {
+			let condaResults = await Promise.all(this.condaLocations.map(location => this.globSearch(location)));
+			let condaFiles = condaResults.reduce((first, second) => first.concat(second));
+			return condaFiles.filter(condaPath => condaPath && condaPath.length > 0);
+		} catch (err) {
+		}
+		return [];
+	}
+
+	private globSearch(globPattern: string): Promise<string[]> {
+		return new Promise<string[]>((resolve, reject) => {
+			glob(globPattern, (err, files) => {
+				if (err) {
+					return reject(err);
+				}
+				resolve(Array.isArray(files) ? files : []);
+			});
+		});
+	}
+
+	private async getPythonSuggestions(): Promise<string[]> {
+		let pathsToCheck = this.getPythonCommands();
+
+		let pythonPaths = await Promise.all(pathsToCheck.map(item => this.getPythonPath(item)));
+		let results: string[];
+		if (pythonPaths) {
+			results = pythonPaths.filter(path => path && path.length > 0);
+		} else {
+			results = [];
+		}
+		return results;
+	}
+
+	private async getPythonPath(options: { command: string; args?: string[] }): Promise<string> {
+		try {
+			let args = Array.isArray(options.args) ? options.args : [];
+			args = args.concat(['-c', '"import sys;print(sys.executable)"']);
+			const cmd = `"${options.command}" ${args.join(' ')}`;
+			let output = await utils.executeBufferedCommand(cmd, {});
+			let value = output ? output.trim() : '';
+			if (value.length > 0 && fs.existsSync(value)) {
+				return value;
+			}
+		} catch (err) {
+			// Ignore errors here, since this python version will just be excluded.
+		}
+
+		return undefined;
+	}
+
+	private getPythonCommands(): { command: string; args?: string[] }[] {
+		const paths = ['python3.7', 'python3.6', 'python3', 'python2', 'python']
+			.map(item => { return { command: item }; });
+		if (process.platform !== constants.winPlatform) {
+			return paths;
+		}
+
+		const versions = ['3.7', '3.6', '3', '2'];
```

Comment:
And/or remove version 2 here so it doesn't show? Our minimum is 3.x, likely 3.5 right?